simulate/simulate.go

```diff
@@ -229,7 +236,20 @@ func MutateGene(inputSeq []dna.Base, branchLength float64, geneFile string) []dn
 		}
 
 	}
-	finalSequence = BaseExtToBases(newSequence)
+	mutatedSequence := BaseExtToBases(newSequence)
+	delFound := seqContainsDels(mutatedSequence)
+	if delFound == false && deletions == true {
+		delEvent := rand.Float64()
+		switch {
+		case delEvent <= branchLength :
+			finalSequence = deleteBase(mutatedSequence)
+		default:
+			finalSequence = mutatedSequence
+		}
+
+	} else {
+		finalSequence = mutatedSequence
+	}
 	return finalSequence
```

Comment:
What happens if the delFound == true && deletions == true? Like if the simulateEvol.go is trying to add a deletion to a gene that already has one. That should return an error to the user, right? Or maybe tell the user, than they can't make a deletion because one already exists? I'm reading line 155 "deletions bool" as a user set feature. 
Or I'm not seeing it and it is already in the code?

Reply:
The intention here is that we only want a given sequence to have at most one deletion. This will give Christi the opportunity to see how single dna.Gap characters behave with her code. If both these variables are true, MutateGene was originally asked to simulate deletions, but the current sequence already has one so we don't want to add more. Eventually, we will remove this check altogether so that deletion mutations will only be controlled by the deletions bool argument.